testing/unittests/jlibtests.cpp

```diff
@@ -3396,7 +3397,629 @@ element1: !el scalarvalue
+
+public:
+    void testSerializeVsSerializeToStreamRootOnlyPTree()
+    {
+        testSerializeVsSerializeToStream(createPTree("EmptyRoot"));
+    }
+
+    void testSerializeVsSerializeToStreamFromXmlPTree()
+    {
+        testSerializeVsSerializeToStream(createPTree(testXml));
+    }
+
+    void testSerializeVsSerializeToStreamFromYamlPTree()
+    {
+        testSerializeVsSerializeToStream(createPTreeFromYAMLString(chkTxt));
+    }
+
+    void testSerializeVsSerializeToStreamCPTValuePTree()
+    {
+        testSerializeVsSerializeToStream(createCPTValuePTree());
+    }
+
+protected:
+    void testSerializeVsSerializeToStream(Owned<IPropertyTree> _originalTree)
+    {
+        originalTree.setown(_originalTree.getClear());
+
+        MemoryBuffer memBuf;
+        originalTree->serialize(memBuf);
+
+        StringBuffer stringStreamBuf;
+        Owned<IBufferedSerialOutputStream> out = createBufferedSerialOutputStream(stringStreamBuf);
+        originalTree->serializeToStream(*out);
+        out->flush();
+        MemoryBuffer memBufStream;
+        memBufStream.append(stringStreamBuf.length(), stringStreamBuf);
+        memBufStream.reset(0);
+        Owned<IBufferedSerialInputStream> in = createMemoryBufferSerialStream(memBufStream);
+        CPPUNIT_ASSERT(in);
+
+        CPPUNIT_ASSERT(SerializeMatches(memBuf, *in, 0));
+    }
+
+    bool readZeroTerminatedString(StringBuffer &out, IBufferedSerialInputStream &in)
+    {
+        out.clear();
+        for (;;)
+        {
+            size32_t got;
+            const char *start = (const char *)in.peek(1, got);
+            if (!start)
+                return false; // eof before nul detected;
+
+            const char *cur = start;
+            const char *end = start + got;
+            while (cur != end)
+            {
+                if (!*cur)
+                {
+                    out.append(cur - start, start);
+                    in.skip((cur - start) + 1); // skip the text and the nul
+                    return true;
+                }
+                cur++;
+            }
+        }
+    }
+
+    bool readByte(byte &out, IBufferedSerialInputStream &in)
+    {
+        size32_t got;
+        const char *start = (const char *)in.peek(1, got);
+        if (!start)
+            return false; // eof before nul detected;
+
+        out = *start;
+        in.skip(1); // skip the byte
+        return true;
+    }
+
+    bool readSize32_t(size32_t &out, IBufferedSerialInputStream &in)
+    {
+        size32_t got;
+        const byte *bytes = (const byte *)in.peek(4, got);
+        if (!bytes)
+            return false; // eof before nul detected;
+
+        in.skip(4); // skip bytes
+        out = ((size32_t)bytes[0]) |
+              ((size32_t)bytes[1] << 8) |
+              ((size32_t)bytes[2] << 16) |
+              ((size32_t)bytes[3] << 24);
+        return true;
+    }
+
+    bool SerializeMatches(MemoryBuffer &memBuf, IBufferedSerialInputStream &in, unsigned depth)
```

Comment:
See comment above - this only needs a memcmp.  You may want to reuse this code for seriailize though.

Also functions should have a lower case starting letter.